### PR TITLE
- added windows support for pushing images to aws ecr

### PR DESCRIPTION
**Pull request by AndreyBulezyuk**
- import platform

## What changes are proposed in this pull request?

Bugfix [2311](https://github.com/mlflow/mlflow/issues/2311). Added support for Windows in the Sagemaker `mlflow sagemaker build-and-push-container` CLI Command.

## How is this patch tested?

Tested on Windows 10, Python 3.6.8 with execution of the code and successful deployment of local ML Model to AWS Sagemaker Service.
A final test on Linux machine would be great. (Can't do that at the moment)

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

(MLFlow Sagemaker CLI now supports sagemaker deployment from a Windows machine.)

### What component(s) does this PR affect?

- [ ] UI
- [x] CLI
- [ ] API
- [ ] REST-API
- [ ] Examples
- [ ] Docs
- [ ] Tracking
- [ ] Projects
- [ ] Artifacts
- [ ] Models
- [ ] Scoring
- [ ] Serving
- [ ] R
- [ ] Java
- [ ] Python

### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [x] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by AndreyBulezyuk** [comment]
Do i have to do something with this PR in order to move it forward? Or do i just wait?

**Comment by mparkhe** [comment]
> @mparkhe Do i have to do something with this PR in order to move it forward? Or do i just wait?

Hi @AndreyBulezyuk: We will take a look at this PR shortly. Thank you for your patience.

**Review by tomasatdatabricks — COMMENTED** [review]
Hi @AndreyBulezyuk, thanks for the PR!

Looks good, I tested it manually on linux and mac and it works as expected. 
I only had one small question.

File: mlflow/sagemaker/__init__.py [hunk 1]
```diff
@@ -127,11 +128,23 @@ def push_image_to_ecr(image=DEFAULT_IMAGE_NAME):
     # x = ecr_client.get_authorization_token()['authorizationData'][0]
     # docker_login_cmd = "docker login -u AWS -p {token} {url}".format(token=x['authorizationToken']
     #                                                                ,url=x['proxyEndpoint'])
-    docker_login_cmd = "$(aws ecr get-login --no-include-email)"
+    os_command_separator = ";\n"
+    if platform.system() == "Windows":
```
[the remark this answers is not in the document]
Comment (AndreyBulezyuk) [reply mlflow/sagemaker/__init__.py]: I have no explicit reason for that, just made most sense. I also saw elsewhere in the code (project/__init__.py) that we use `if os.name != "nt":` to determine whether we are on windows.

I think i should research a bit more and find a more scalable way to figure out on what OS we are. Maybe i should take a look at how tensorflow or other major python projects do that?

[the remark this answers is not in the document]
Comment (AndreyBulezyuk) [reply mlflow/sagemaker/__init__.py]: Seems like tensorflow uses 'platform' quite often. Here their a simple method for windows identification:

```
def _is_windows():
  return platform.system() == "Windows"
```

Not sure if mlflow has a global repository of methods/helpers, but we could code a few OS identifying methods (like the one above) and use these helpers everywhere else in the code. This way if we decide to change the way we identify a OS, we must do so only in one location (helper/global methods).

How does that sound?

https://github.com/tensorflow/tensorflow/blob/b04371bc952f9f9668e862d82db71651fdef8dc6/third_party/gpus/check_cuda_libs.py#L42

[the remark this answers is not in the document]
Comment (tomasatdatabricks) [reply mlflow/sagemaker/__init__.py]: Sounds great! Thanks for looking into it. I think we can do that in a followup though and merge this PR as is.

[the remark this answers is not in the document]
Comment (AndreyBulezyuk) [reply mlflow/sagemaker/__init__.py]: Agree :)

[the remark this answers is not in the document]
Comment (tomasatdatabricks) [reply mlflow/sagemaker/__init__.py]: Awesome, merging now, thanks again for the contribution!

**Review by tomasatdatabricks — APPROVED** [review]
lgtm

File: mlflow/sagemaker/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (AndreyBulezyuk) [reply mlflow/sagemaker/__init__.py]: Are you in mlflow Slack? @tomasatdatabricks

[the remark this answers is not in the document]
Comment (tomasatdatabricks) [reply mlflow/sagemaker/__init__.py]: Yes I am (name is Tomas Nykodym).

**Pull request merged**